subprojects/gradle/tms/src/main/kotlin/com/avito/plugin/MarkReportAsSourceAction.kt

```diff
@@ -26,10 +25,10 @@ class MarkReportAsSourceAction(
     fun mark(reportCoordinates: ReportCoordinates) {
         val testSuiteVersion = timeProvider.nowInMillis()
 
-        when (val result = reportsApi.getReport(reportCoordinates)) {
-            is GetReportResult.Found ->
+        reportsApi.getReport(reportCoordinates).fold(
```

Comment:
I see result API as:
getReport
.flatMap { report -> }
.onError { error -> }

Reply:
This is out of scope but it will improve readability

Reply:
`Result<T>.flatmap()` is used to return another `Result<R>`, and you want it to return `T` here
fold on the other hand returns T

maybe
```
getReport
.onSuccess { report -> }
.onError { error -> }
```

is what you want? just plain side effects expression

Reply:
Btw, why don't we use [this](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/) implementation of Result? Seems like it solves the same issues

Reply:
>Result<T>.flatmap() is used to return another Result<R>

Sorry but I see you make another request on Success for is a `flatmap`

Reply:
> Btw, why don't we use this implementation of Result? Seems like it solves the same issues

- It's experimental and could be somehow changed in future releases
- it requires to enable special compiler option to go without error
- it sometimes exposes in modules API, which will require clients to enable this compiler options

Reply:
> Sorry but I see you make another request on Success for me it is a flatmap

ah, now i understand. let's see

Reply:
#936

Reply:
> why don't we use this implementation of Result? Seems like it solves the same issues
> ...

You discussed the reasons. For others, they will remain in this thread.
Let's fix knowledge in `Result` documentation. 
Also, it will help to understand when to switch to Kotlin's Result.

Reply:
#944